service-deployment/deploy.py

```diff
@@ -191,6 +206,33 @@ def generate_template_file(cluster_config, service_config):
 
 
 
+def single_service_bootstrap(serv, service_config):
```

Comment:
>single_service_bootstrap [](start = 4, length = 24)

what is the difference from dependency_bootstrap? do you need to consider service dependency here?

Reply:
Single service startup will ignore the dependency. User should ensure their operation is in a correctly order.

Reply:
dependency_bootstrap will auto-start all service in a correctly order.

Reply:
`single_service_bootstrap` can also be reused in `bootstrap_service`?

Reply:
got it. please add a comment here.

---
In reply to: [159383031](https://github.com/Microsoft/pai/pull/159#discussion_r159383031) [](ancestors = 159383031)

Reply:
@abuccts @fanyangCS  should we run a stop script in a function?

Reply:
If we reuse single startup in bootstrap_service. If user mis-use our command. Then all service will be redeploy. In current process, if service already exists in the cluster, re-executing the startscript won't have any influence. And that's the one reason why i will make the function single_service_bootstrap  ignore dependency.

Reply:
I understand you introduce this function for debug purpose. I prefer everything done manually in this case.

---
In reply to: [159384788](https://github.com/Microsoft/pai/pull/159#discussion_r159384788) [](ancestors = 159384788)